### ISSUE TITLE
v4 and up causes Aborted Connection errors when uploading images

### ISSUE THREAD

**Issue by evoLucidity** [docker-library/ghost#266]
I'm running Docker version 20.10.8, build 3967b7d on Ubuntu Server 20.04.

Versions 3.42.5 and below of Ghost work fine in Docker, but I have problems with v4 and above.

When I try to up load an image (post image, or publication icon etc) I see "Request was rejected due to server error" on the Ghost admin panel and:

```
ghost         | [2021-08-29 18:04:50] INFO Scheduling job update-check at 54 0 21 * * *. Next run on: Sun Aug 29 2021 21:00:54 GMT+0000 (Coordinated Universal Time)
ghost         | [2021-08-29 18:05:04] INFO "GET /ghost/" 200 54ms
ghost         | [2021-08-29 18:05:04] INFO "GET /ghost/api/canary/admin/users/me/?include=roles" 200 61ms
ghost         | [2021-08-29 18:05:04] INFO "GET /ghost/api/canary/admin/site/" 200 7ms
ghost         | [2021-08-29 18:05:04] INFO "GET /ghost/api/canary/admin/config/" 200 13ms
ghost         | [2021-08-29 18:05:05] INFO "GET /ghost/api/canary/admin/settings/?group=site%2Ctheme%2Cprivate%2Cmembers%2Cportal%2Cnewsletter%2Cemail%2Camp%2Clabs%2Cslack%2Cunsplash%2Cviews%2Cfirstpromoter%2Coauth%2Ceditor" 200 31ms
ghost         | [2021-08-29 18:05:05] INFO "GET /ghost/api/canary/admin/notifications/" 200 27ms
ghost         | [2021-08-29 18:05:05] INFO "GET /ghost/api/canary/admin/members/stats/count/" 200 147ms
ghost         | [2021-08-29 18:05:06] INFO "GET /favicon.png" 200 3ms
ghost         | [2021-08-29 18:05:06] INFO "POST /" 200 503ms
ghost         | [2021-08-29 18:05:06] INFO "GET /assets/built/screen.css?v=dd2bba50af" 200 4ms
mariadb       | 2021-08-29 18:05:06 68 [Warning] Aborted connection 68 to db: 'ghost_blog_prod' user: 'ghostbloguser' host: '172.24.0.6' (Got an error reading communication packets)
mariadb       | 2021-08-29 18:05:06 66 [Warning] Aborted connection 66 to db: 'ghost_blog_prod' user: 'ghostbloguser' host: '172.24.0.6' (Got an error reading communication packets)
mariadb       | 2021-08-29 18:05:06 67 [Warning] Aborted connection 67 to db: 'ghost_blog_prod' user: 'ghostbloguser' host: '172.24.0.6' (Got an error reading communication packets)
mariadb       | 2021-08-29 18:05:06 65 [Warning] Aborted connection 65 to db: 'ghost_blog_prod' user: 'ghostbloguser' host: '172.24.0.6' (Got an error reading communication packets)
mariadb       | 2021-08-29 18:05:06 64 [Warning] Aborted connection 64 to db: 'ghost_blog_prod' user: 'ghostbloguser' host: '172.24.0.6' (Got an error reading communication packets)
ghost exited with code 132
ghost         | [2021-08-29 18:05:08] INFO Ghost is running in production...
ghost         | [2021-08-29 18:05:08] INFO Your site is now available on https://blog.simons.earth/
ghost         | [2021-08-29 18:05:08] INFO Ctrl+C to shut down
ghost         | [2021-08-29 18:05:08] INFO Ghost server started in 0.58s
ghost         | [2021-08-29 18:05:09] INFO Database is in a ready state.
ghost         | [2021-08-29 18:05:09] INFO Ghost database ready in 1.342s
ghost         | [2021-08-29 18:05:12] INFO Ghost booted in 4.874s
ghost         | [2021-08-29 18:05:12] INFO Adding offloaded job to the queue
ghost         | [2021-08-29 18:05:13] INFO Scheduling job update-check at 51 21 14 * * *. Next run on: Mon Aug 30 2021 14:21:51 GMT+0000 (Coordinated Universal Time)
mariadb       | 2021-08-29 18:05:16 83 [Warning] Aborted connection 83 to db: 'ghost_blog_prod' user: 'ghostbloguser' host: '172.24.0.6' (Got an error reading communication packets)
mariadb       | 2021-08-29 18:05:16 82 [Warning] Aborted connection 82 to db: 'ghost_blog_prod' user: 'ghostbloguser' host: '172.24.0.6' (Got an error reading communication packets)
mariadb       | 2021-08-29 18:05:16 81 [Warning] Aborted connection 81 to db: 'ghost_blog_prod' user: 'ghostbloguser' host: '172.24.0.6' (Got an error reading communication packets)
mariadb       | 2021-08-29 18:05:16 80 [Warning] Aborted connection 80 to db: 'ghost_blog_prod' user: 'ghostbloguser' host: '172.24.0.6' (Got an error reading communication packets)
mariadb       | 2021-08-29 18:05:16 79 [Warning] Aborted connection 79 to db: 'ghost_blog_prod' user: 'ghostbloguser' host: '172.24.0.6' (Got an error reading communication packets)
```

in the Docker log.

MariaDB version is 'latest'. I see the same result with MySQL.

What am I missing, please?

**Comment by pascalandy** [comment]
Check on Ghost's forums which DB is supported. It would also help if you share your compose file. Cheers!

**Comment by acburdine** [comment]
my guess is this is related to #256 - based on the "when I upload image, Ghost crashes" error.

**Comment by evoLucidity** [comment]
> my guess is this is related to #256 - based on the "when I upload image, Ghost crashes" error.

Reading through the thread, I agree it looks very similar.
I'll try later with later alpine versions.

**Comment by evoLucidity** [comment]
Tried running with NODE_ENV: development
```
steve@webserver:~/docker-files$ docker exec ghost node --version
v14.17.5
steve@webserver:~/docker-files$ docker exec ghost ghost --version
- Inspecting operating system
- Checking for deprecations
Ghost-CLI version: 1.17.3
Ghost version: 4.12.1 (at /var/lib/ghost)
```
Same results as before.

Relevant sections from docker-compose.yml:
```
  mariadb:
    image: mariadb:latest
    restart: always
    container_name: mariadb
    volumes:
      - mariadb-data:/var/lib/mysql
    ports:
      - 3306:3306
    networks:
      - web
    environment:
      MYSQL_ROOT_PASSWORD: [redacted]
      MYSQL_DATABASE: ghost_blog_prod
      MYSQL_USER: "ghostbloguser"
      MYSQL_PASSWORD: "[redacted]"
      MYSQL_ROOT_USER: "root"

  ghost:
    image: ghost:alpine
    container_name: ghost
    restart: always
    environment:
      database__client: mysql
      database__connection__host: mariadb
      database__connection__user: ghostbloguser
      database__connection__password: [redacted]
      database__connection__database: ghost_blog_prod
      url: https://blog.simons.earth
      NODE_ENV: development
    ports:
      - 2368:2368
    volumes:
      - ghost-data:/var/lib/ghost/content
    depends_on:
      - mariadb
    networks:
      - web

volumes:
  portainer_data: {}
#  mysql-data: {}
  mariadb-data: {}
  ghost-data: {}
  dokuwiki-data: {}
  apache-data: {}
```

**Comment by evoLucidity** [comment]
I was originally running this on a DO droplet and it was working okay.
My issues started when I transferred the server to a spare box in my house.
I've just moved to a Hetzner box and it's working fine on the same OS with the same versions of Ghost and Node.

So it's likely a network issue causing my problems.
Since I'm now unable to to create the issue I'm closing the issue. 

Thank you for your time.
